### DISCUSSION TITLE
JSON file structure for Snippets Extension

### DISCUSSION THREAD

**Discussion by allordacia** [microsoft/vscode-discussions#255] [Extension Development QnA]
I have been working on a snippets extension that would provide snippets/suggestions based on looking up data stored in three large JSON files included in the extension directory. I have been successful getting it to work at a basic level however there is a serious delay when the lookup trigger is typed in addition to also be sometimes searching even when the prompt isnt provided which makes me think Ive got my JSON files structured incorrectly. Hoping that someone could confirm this is the correct or proper way to organize my snippets or point out where I went wrong with it.

Each JSON object in the files are structured like this:

```
{
		"1426": { 
			"prefix": ["Broker Fee Line 1"],
			"body": ["[${1:1426}] = ${2:}"],
			"description": ""
		}
}
```
Obviously, comas separating each object with unique data to match the thousands of fields in each JSON file. Where I am getting confused by how the Snippets guide explains organizing these JSON files, and it is definitely a misunderstanding on my part, is which data pieces are the ones that _should_ trigger the lookup to be triggered? Does the name of the Snippet (In this case "1426") what should trigger the lookup or should that only happen when they would begin to type the prefix information? There are a lot of objects in my files that have identical prefix or Snippet names for the first part of their values and are only unique by the last character in their name or prefix/description, how do I either restrict the number of options that are returned to the user or increase how much information is required to be typed before the object is returned as an option? 

I feel like there is possibly a better way to organize this to make it more manageable to use with my projects and I am curious to hear what suggestions others have to help me fix what I am doing wrong!

**Comment by isidorn** [comment]
These docs should help https://code.visualstudio.com/docs/editor/userdefinedsnippets
If they do not let us know what is the exact question we can help with. Thanks

**Reply by allordacia** [reply]
Sorry for the late reply, this does not help answer the question as it doesnt explain how best to organize or parse keywords from such a large database of snippets.

**Reply by burkeholland** [reply]
I don't know that we provide this guidance. I'd be curious as to what your experience was and possibly something we could not document for the community here. Closing this issue for now due to date, but please please re-open if you were able to work out an answer.

**Reply by pkdelli** [reply]
Hi, I am building a VS Code Snippets extension. one of the JSON files in the snippets folder is more than 100kb and vsce package/build process raised a flag saying "The file extension/snippets/templates.json is large (114.85 KB)". Wondering if it is a best practice to keep the individual snippet JSON files under 100kb. How many such JSON files are too much. Can I have 10 snippet JSON files, each  under 100kb?
